students/bbirsoz1/Module_5/mailroom3.py

```diff
@@ -0,0 +1,126 @@
+
+def create_a_report():
+    print("Report:")
+    Header = ('Donor Name', 'Total Donation', "Number of Donations", 'Average Donation')
+    print("-"*90)
+    print("{:<20} | {:^24} | {:^19} | {:^15}".format(Header[0], Header[1], Header[2], Header[3]))
+    print("-"*90)
+
+    everything = [ ]
+    #everything = [donor_db.items]
+    for donor_name, donations in donor_db.items():
+        total_donation = sum(donations)
+        num_of_donation = len(donations)
+        average_donation = int(total_donation / num_of_donation)
+        print("{:<20} | {:>24} | {:^19} | {:>15}".format(donor_name, total_donation, num_of_donation, average_donation))
+    print("-"*90)
+
+def mail_donor():
+    list_of_donors()
+    current_d = "" #donors currently listed
+    current_d = str(input("Who would you like to mail?:"))
+    if current_d in donor_db:
+        add_donation(current_d)
+    else:
+        donor_db[current_d] = []
+        add_donation(current_d)
+
+def add_donation(current_d):
+    donation_times = float(input("How many donations have been made so far?: ")) #replaced int() with float()
+    #if donation_times.isdigit(): #why does this not work instead of while true?
+    while True:
+        try:
+            donation_times = int(donation_times)
+            break
+        except ValueError:
+            print("Input must be an integer, try again.") #it doesnt show up, why???
+            break
+    total_donation = 0
+    while donation_times > 0:
+        new_donation = float(input("Donation amount?: "))
+        donor_db[current_d].append(new_donation)
+        donation_times -= 1
+        total_donation += float(new_donation)
+    mail_note(current_d, total_donation)
+    
+def mail_note(donor_name, total_donation):
+    print(f"Thank you {donor_name} for your very kind donation of ${total_donation}")
+
+def send_letters_to_all():
+        for key in donor_db:
+            with open(key+"_"+str(datetime.date.today())+".txt", 'w') as f:
+                f.write("Dear {fullname},\n"
+                        "Thank you for your very kind donation of ${amount:10.2f}."
+                        " It will be put to very good use.\n"
+                        "Sincerely,\n"
+                        "~The Team".format(fullname=key, amount=donor_db[key][0]))
+        print("Thank you letters to all donors have been saved to the local directory.\n")
+
+def quit_program():
+    print("Goodbye!")
+    sys.exit()
+
+def thank_menu():
+    entry=("1","2","3")
+    while True:
+        mail = input(thank_menu_prompt)
+        if mail not in entry:
+            print("Not a valid option!")
+        elif mail == "1":
+            mail_donor()
+        elif mail == "2":
+            list_of_donors()
+        elif mail == "3":
+            break
+
+def main():
+    while True:
+        response = int(input(main_menu_prompt))
```

Comment:
Add exception handling here as well. It warns you if you enter a number that doesn't match any menu items, but entering a string crashes your code.